### PR TITLE
Simple golfclub that makes and jumps to balls

### PR DESCRIPTION
**Pull request by PhilipRosedale**
Simple golfclub (putter) that drops balls you can hit to play things like mini-golf.   To use: 

import the file golfClub.json to make a club, or you can attach  golfclub.js to a different club model.   If you do the latter, make sure you have a collision hull so that it will hit balls.  

1.  Equip the club and squeeze trigger to drop a new ball at your feet. 
2.  Hit ball!
3.  To jump to ball, squeeze trigger.  If ball has gotten too far away, new ball will drop.

### CODE REVIEW

**Comment by PhilipRosedale** [comment]
Test plan:  Follow directions above.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 61f4009b22ba9687d11b95595336fe897deaf006 available at: https://gist.github.com/8fb49277462d69468800722551090644<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7616.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7616.dmg

File: examples/golfclub/golfClub.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
+//
+//  golfClub.js
+//
+//  Created by Philip Rosedale on 11/11/15.
```
Comment (thoys) [inline examples/golfclub/golfClub.js]: Are you sure about the date?

**Comment by thoys** [comment]
## QA results

- works great once I was able to grab the club :+1: 
- ball gets created at feet on initial squeeze while held :+1: 
- when the ball is close it teleports you to it after squeezing the hydra's. :+1: 
- new ball gets created when ball too far away or lost. :+1: 

Test pass!

Coincidence: I was working on almost the same thing as you today. Was working on a club/ball rezzer at caitlyn's minigolf. hehe :+1: 

Tip: Maybe you can add some gravity to the club entity JSON?

File: examples/golfclub/golfClub.js (hunk 1 above)
Comment (ZappoMan) [inline examples/golfclub/golfClub.js]: Apparently he's been working on this in secret for months!!

File: examples/golfclub/golfClub.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+    "Entities": [
+        {
+            "collisionsWillMove": 1,
+            "compoundShapeURL": "http://hifi-content.s3.amazonaws.com/caitlyn/dev/Tutorials/Putter_phys.obj",
```
Comment (howard-stearns) [inline examples/golfclub/golfClub.json]: Let me see if I can make you a valid .obj file. This one isn't, and there's going to be problems if someone uses it as a renderable model.

File: examples/golfclub/golfClub.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    return Vec3.sum(MyAvatar.position, Vec3.multiply(BALL_DROP_DISTANCE, Quat.getFront(MyAvatar.orientation)));
+}
+
+function avatarHalfHeight() {
+    return MyAvatar.getDefaultEyePosition().y - MyAvatar.position.y;
+}
+
+function brakeBall(ball) {
+    //  Check the ball's velocity and slow it down if beyond a threshold
+    var properties = Entities.getEntityProperties(ball);
```
Comment (howard-stearns) [inline examples/golfclub/golfClub.js]: Should this be Entities.getEntityProperties(ball, ['velocity'])?
I wonder if we should have a coding standard on whether our examples should enumerate the properties they want to look at. I.e., are we accumulating a performance debt?

File: examples/golfclub/golfClub.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        triggerState = true;
+    }  else if (triggerValue < TRIGGER_RELEASE_THRESHOLD) {
+        triggerState = false;
+    }
+    return (triggerState && (oldTriggerState != triggerState)); 
+}
+
+function ballPosition(ball) {
+    //  return the position of this entity
+    var properties = Entities.getEntityProperties(ball);
```
Comment (howard-stearns) [inline examples/golfclub/golfClub.js]: Same question as for velocity, below. Here it's for 'position'.

File: examples/golfclub/golfClub.json (hunk 2 above)
Comment (PhilipRosedale) [inline examples/golfclub/golfClub.json]: OK - addressed the two code review points.  @howard-stearns, can you fix this with Caitlyn and then test/pull this PR?

**Comment by hifi-gustavo** [comment]
Build console output for commit: aab93d8a267824928f34dba969ea3a750130d192 available at: https://gist.github.com/8fb49277462d69468800722551090644<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7616.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7616.dmg

File: examples/golfclub/golfClub.json (hunk 2 above)
Comment (howard-stearns) [inline examples/golfclub/golfClub.json]: Here's a corrected .obj file: http://howard-stearns.github.io/models/Putter_phys.obj
Please put it in s3 in either the same place. (If put in a different place, don't forget to update the reference in golfClub.json.)

**Comment by pippilina** [comment]
I've updated the putter collider per your request @howard-stearns

**Pull request merged**